### PR TITLE
Complete the need-to-modify/unimplemented APIs to align with the specification

### LINKED ISSUE #157: Align send() API with specification
Current `send()` implementation does not support the following parameters for a message with value transaction 
1. output (Users can manually pick their own output instead of having node decide on which output should be used.)
2. indexation_key (An optional indexation key of the indexation payload.)
3. data (An optional indexation data of the indexation payload.)

Also, current `output` parameter in the `send()` is actually the `address` parameter defined in the [spec](https://github.com/iotaledger/iota.rs/blob/dev/specs/iota-rs-ENGINEERING-SPEC-0000.md), which is needed to be rename as `address`.

### PR DESCRIPTION
**Pull request by bingyanglin**
Fix #157 #158

### CODE REVIEW

File: iota-client/src/client.rs [hunk 1]
```diff
@@ -175,6 +175,29 @@ impl Client {
...
+        for output in outputs {
+            let meta_data = self.get_output(output).await?;
+            output_metadata.push(meta_data);
+        }
+
+        // Use `get_address()` API to get the address outputs first, then use the `get_output()`
+        // to get the `OutputMetadata`
+        for address in addresses {
+            let address_outputs = self.get_address().outputs(&address).await?;
+            for output in address_outputs.iter()  {
```
Comment (lucasfernog) [inline iota-client/src/client.rs]: should we check if the `outputs` parameter contains the output here to prevent duplication?

Comment (bingyanglin) [reply iota-client/src/client.rs]: Good point. Will use a `HashSet` to collect all the queried outputs first. Thank you for pointing this out.

**Review by lucasfernog — APPROVED** [review]

**Comment by lucasfernog** [comment]
@bingyanglin can you comment out the `with_network_id` call? it was temporarily removed from bee.

**Comment by lucasfernog** [comment]
LGTM? can't wait to implement this on the wallet lib :D

**Pull request merged**